### ISSUE TITLE
Crashing Xcode 7.3.1/macOS Sierra DVTSourceTextView setTextContainer

### ISSUE THREAD

**Issue by devandanger** [CocoaPods/CocoaPods#5988]
# Report
OK, I know this sounds totally unrelated but I was able to reproduce this issue twice.  I added a pod to my unit testing target and started getting what I felt was a totally unrelated crash when I then entered my text editor in Xcode.  The crash referred to the following error (DVTSourceTextView setTextContainer, objc_retain) Scratched my head on this for awhile thought it might be compatibility issues with Sierra.  Opened the xcodeproj instead and actually could access the text editor again.  Opened the xcworkspace same crash.  Deleted  the workspace/Pods, pod install.  Xcworkspace works again.  Checked in moved on with my life, out of curiosity went back to older checkin crashed again.  Seems like it is a reproducible problem in general assuming it was something the tool did aside from something I did wrong.

I consider this matter closed in general, but worth having some documentation in case someone else has the problem and wants to try this hail mary.  If you guys are interested in pursuing a true root cause, I'd be happy to help otherwise feel free to CLOSE this issue.

## What did you do?
Opened up an xcode workspace setup with cocoapods (1.0.1) in Xcode 7.3.1 and received a crash anytime I access a file for text editing usage.

## What did you expect to happen?
I expected the text editor to open showing lines of code.
e.g. Install all pod dependencies correctly.

## What happened instead?
Crashed :(

## CocoaPods Environment
1.0.1

**Comment by benasher44** [comment]
You could try `pod deintegrate`, and then `pod install` again to see if that helps.

**Comment by ricobeck** [comment]
Had the same issue and figured out that the workspace package was corrupt. After `pod deintegrate`, deletion of workspace, and `pod install` it worked again.

**Comment by devandanger** [comment]
Gotcha, pod deintegrate works better.

**Comment by alonre** [comment]
Not working for us (CocoaPods 1.1.0.rc.3 + Xcode 7.3.1).
Xcode still crashes after `pod deintegrate` and `pod install` (several attempts).
A computer reboot will temporarily solve the problem, but not for long.
Maybe some cache corruption?

**Comment by devandanger** [comment]
Noticed some white noise regarding having multiple xcodes installed (7.3.1 and 8).  I noticed the error when switching between versions opening the same project (this project didn't even use cocoapods though).  Is this is part of your scenario?

**Comment by alonre** [comment]
You nailed it.
We had to have several versions of the Platform SDK to build OSXFUSE (FUSE for macOS).
However, troubles started only for one specific workspace, and only after we integrated CocoaPods into it.
#### TLDR:
I'll test and update later today.
Thanks!

**Comment by benasher44** [comment]
According to [this SO post](http://stackoverflow.com/questions/38903699/xcode-7-3-1-crashes-on-macos-sierra-beta-5), this looks to be a macOS Sierra compatibility issue with Xcode 7.3.1. It does say in the release notes for macOS Sierra beta 5 that Xcode 7.3.1 is not compatible with macOS Sierra. I'm going to close this, but please open a new issue if people find that this is still an issue with Xcode 7.3.2/8.0.

**Comment by akaralar** [comment]
In my case deleting each log in `~/Library/Developer/Xcode/iOS Device Logs/` solved the problem. @devandanger thanks for the cue! It started to happen with me after I built it with xcode 8, it might be that xcode 7 is maybe trying to parse logs of xcode 8 which are not backwards compatible? I don't know but it solved my problem.
